CONTRIBUTING.md

```diff
@@ -0,0 +1,222 @@
+# Contributing to elixir-nats
+
+Please take a moment to review this document in order to make the contribution
+process easy and effective for everyone involved!
+
+## Using the issue tracker
+
+Use the issues tracker for:
+
+* [bug reports](#bug-reports)
+* [submitting pull requests](#pull-requests)
+
+Please **do not** use the issue tracker for personal support requests nor feature requests. Instead,
+please reach out to us at
+
+* [info@nats.io](mailto:info@nats.io)
+* [Twitter](https://twitter.com/nats_io)
+* [Google Mailing List](https://groups.google.com/forum/#!forum/natsio)
+* [Slack](https://docs.google.com/a/apcera.com/forms/d/104yA7oqq7SPoMDG_J9MnVE74gVwBnTmVHKP5ABHoM5k/viewform?embedded=true)
+
+
+## Bug reports
+
+A bug is a _demonstrable problem_ that is caused by the code in the repository.
+Good bug reports are extremely helpful - thank you!
+
+Guidelines for bug reports:
+
+1. **Use the GitHub issue search** &mdash; [check if the issue has already been
+   reported](https://github.com/nats-io/elixir-nats/search?type=Issues)
+
+2. **Check if the issue has been fixed** &mdash; try to reproduce it using the
+   `master` branch in the repository.
+
+3. **Isolate and report the problem** &mdash; ideally create a reduced test
+   case.
+
+Please try to be as detailed as possible in your report. Include information about
+your Operating System, your Erlang and Elixir versions. Please provide steps to
+reproduce the issue as well as the outcome you were expecting! All these details
+will help developers to fix any potential bugs.
+
+Example:
+
+> Short and descriptive example bug report title
+>
+> A summary of the issue and the environment in which it occurs. If suitable,
+> include the steps required to reproduce the bug.
+>
+> 1. This is the first step
+> 2. This is the second step
+> 3. Further steps, etc.
+>
+> `<url>` - a link to the reduced test case (e.g. a GitHub Gist)
+>
+> Any other information you want to share that is relevant to the issue being
+> reported. This might include the lines of code that you have identified as
+> causing the bug, and potential solutions (and your opinions on their
+> merits).
+
+## Feature requests
+
+Feature requests are welcome and should be discussed on [the natsio mailing list](https://groups.google.com/forum/#!forum/natsio),
+or [reach out on Slack](https://docs.google.com/a/apcera.com/forms/d/104yA7oqq7SPoMDG_J9MnVE74gVwBnTmVHKP5ABHoM5k/viewform?embedded=true).
+
+## Contributing
+
+We invite everyone to contribute to *elixir-nats* and help us tackle
+existing issues!
+
+With tests running and passing, you are ready to contribute and
+send your pull requests.
+
+## Contributing Documentation
```

Comment:
Suggest we refer to an existing project for contribution guidelines that are _standardized_ such as pull/merge semantics working on branches, documentation conventions and guidelines. Prefer we refer to best cases rather than write these up.